pkg/resourcesets/collector.go

```diff
@@ -187,47 +187,35 @@ func (h *ResourceHandler) gatherResourcesForGroupVersion(filter v1.ResourceSelec
 }
 
 func (h *ResourceHandler) gatherObjectsForResource(ctx context.Context, res k8sv1.APIResource, gv schema.GroupVersion, filter v1.ResourceSelector) ([]unstructured.Unstructured, error) {
-	var filteredByNamespace, filteredObjects []unstructured.Unstructured
+	var filteredByNamespace []unstructured.Unstructured
 	gvr := gv.WithResource(res.Name)
-	var dr dynamic.ResourceInterface
-	dr = h.DynamicClient.Resource(gvr)
+	dr := h.DynamicClient.Resource(gvr)
 
 	// only resources that match name+namespace+label combination will be backed up, so we can filter in any order
 	// however, in practice filtering by label happens at an API level when we paginate the resources (creating our initial list)
 	filteredByLabel, err := h.filterByLabel(ctx, dr, filter)
 	if err != nil {
-		return filteredObjects, err
+		return nil, err
 	}
 
-	filteredByLabelItems := h.appendTempUID(filteredByLabel.Items)
-	filteredByName, err := h.filterByName(filter, filteredByLabelItems)
+	filteredByName, err := h.filterByName(filter, &filteredByLabel.Items)
```

Comment:
I don't see why `filtererByLabel` needs to be passed by reference, as its contents are modified by the callee, but not the wrapper itself.